### PR TITLE
Add cross references to manual

### PR DESCRIPTION
**Pull request by aadcg**
# Description

@aartaka had introduced this in c8b76a092686e053de08602b4d78c0c4095c61bd. I started doing something similar last week. But I wanted to make it a bit more general.

# Checklist:
Everything in this checklist is required for each PR.  Please do not approve a PR that does not have all of these items.  

- [x] I have pulled from master before submitting this PR
- [x] My code follows the style guidelines for Common Lisp code
  - See [Norvig & Pitman's Tutorial on Good Lisp Programming Style (PDF)](https://www.cs.umd.edu/~nau/cmsc421/norvig-lisp-style.pdf) and [Google Common Lisp Style Guide](https://google.github.io/styleguide/lispguide.xml).
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer (the peer review to approve a PR counts.  The reviewer must download and test the code)
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

**Comment by aadcg** [comment]
I tried to re-invent the wheel when `spinneret` already provided a way to write macros! I should have read its manual first.

The use of `:h*` would be elegant, but unfortunately it doesn't suit our needs. Notice that:

```lisp
(spinneret:with-html-string
  (:section (:h* "test")))
```

generates:

```html
<section>
 <h2>test</h2>
</section>
```

Which means that we wouldn't be able to cross-reference h1 sections.

**Comment by aartaka** [comment]
There should be only one `h1` tag, and it should match the page title, so not much use in referencing it :)

**Comment by aadcg** [comment]
> There should be only one h1 tag

I see.

Should be ready to merge.

File: source/tutorial.lisp [hunk 1]
```diff
@@ -5,363 +5,381 @@
...
+      (list-command-information '(nyxt/password-mode:copy-username nyxt/password-mode:copy-password nyxt/password-mode:copy-password-prompt-details)))
+     (:p "And more: ")
+     (:ul
+      (list-command-information '(nyxt/document-mode:paste-from-clipboard-ring show-system-information))))
+
+   (:nsection :title "Link navigation"
+     (:p "Link-hinting allows you to visit URLs on a page without using the mouse.
 Invoke one of the commands below: several hints will appear on screen and all
 links on the page will be listed in the prompt buffer.  You can select the hints
 by matching against the hint, the URL or the title.")
```
Comment (aartaka) [inline source/tutorial.lisp:145]: While we're at it, I'd mention that hints also allow hovering, focusing, etc.

File: source/tutorial.lisp [hunk 2]
```diff
@@ -5,363 +5,381 @@
...
+                                  nyxt/bookmark-mode:bookmark-url nyxt/bookmark-mode:bookmark-hint
+                                  nyxt/bookmark-mode:set-url-from-bookmark nyxt/bookmark-mode:delete-bookmark
+                                  nyxt/bookmark-mode:list-bookmarks
+                                  nyxt/bookmark-mode:import-bookmarks-from-html
+                                  nyxt/bookmark-frequent-visits:bookmark-frequent-visits-mode))))
+
+   (:nsection :title "Annotations"
+     (:p "Annotations can have the following settings:")
+     (:ul
+      (:li (:code ":snippet") ": The snippet which was highlighted by the user.")
```
Comment (aartaka) [inline source/tutorial.lisp:197]: Maybe even reference the slots with `:nxref`?

**Review by aartaka — APPROVED** [review]

File: source/manual.lisp [hunk 3]
```diff
@@ -11,65 +11,61 @@
      (:p "This manual first includes the tutorial, then covers the configuration
 of Nyxt."))
    (tutorial-content)
-   (config-content)))
+   (manual-sections)))
 
-(defun config-content ()
+(defun manual-sections ()
```
[the remark this answers is not in the document]
Comment (aadcg) [reply source/manual.lisp:16]: It doesn't describe configuration alone. It's much broader than that.

File: source/manual.lisp [hunk 4]
```diff
@@ -11,65 +11,61 @@
...
-                                        (echo "Configuration file exists")
-                                        (progn (ensure-directories-exist config-file-path)
-                                               (ensure-file-exists config-file-path)
-                                               (echo "Configuration file created at ~s."
-                                                     config-file-path)))))
-                      "Create configuration file"))))))
-
-     (:p "Example:")
-     (:pre (:code "(define-configuration buffer
+             (let ((config-file-path (files:expand *config-file*)))
```
[the remark this answers is not in the document]
Comment (aadcg) [reply source/manual.lisp:45]: That functionality is there, but it's broken. 

Anyway it's only useful for those who start Nyxt from the REPL, and those clearly know how to create a file :D

[the remark this answers is not in the document]
Comment (aadcg) [reply source/manual.lisp:45]: Wait. I misinterpreted some things.

File: source/manual.lisp (hunk 3 above)
[the remark this answers is not in the document]
Comment (aadcg) [reply source/manual.lisp:16]: `manual-content` is the function that calls `config-content/manual-sections` :upside_down_face:

File: source/manual.lisp [hunk 5]
```diff
@@ -274,7 +256,7 @@ specify which input and output values are expected.")
                 ") as a shorthand for the " (:code "nhooks:add-hook") ".")
            (:li (:nxref :function 'nhooks:remove-hook) " (also available as "
                 (:code "hooks:remove-hook") ") that removes the handler from a certain hook.")
-           (:li (:code "hnooks:once-on") " (also available as " (:code "hooks:once-on")
+           (:li (:code "nhooks:once-on") " (also available as " (:code "hooks:once-on")
```
[the remark this answers is not in the document]
Comment (aadcg) [reply source/manual.lisp:259]: I should have separated this commit but I still didn't find an effortless way of doing so :D

File: source/tutorial.lisp [hunk 6]
```diff
@@ -5,363 +5,381 @@
 
 (defun tutorial-content ()
   (spinneret:with-html-string
-   (:h2 "Core concepts")
-   (:h3 "Keybindings and commands")
-   (:p "Commands are invoked by pressing specific keys or from
+   (:nsection :title "Core concepts")
```
[the remark this answers is not in the document]
Comment (aadcg) [reply source/tutorial.lisp:8]: All mistakes. The same applies to the manual.

[the remark this answers is not in the document]
Comment (aadcg) [reply source/tutorial.lisp]: Fixed.

File: source/manual.lisp [hunk 7]
```diff
@@ -465,11 +435,27 @@ input area containing familiar code, with some " (:code "v332 = \"Hello, Nyxt!\"
 " variable assignment, and with a verbatim text outputted by your code:")
       (:pre (:code "Hello, Nyxt!"))
       (:p "This cell-based code evaluation is the basis of the Nyxt REPL. For more features, see "
-          (:nxref :package :nyxt/repl-mode "REPL mode documenation") "."))
+          (:nxref :package :nyxt/repl-mode "REPL mode documentation") "."))
+
+    (:nsection :title "Built-in REPL (" (:nxref :package :nyxt/repl-mode) ")"
```
[the remark this answers is not in the document]
Comment (aadcg) [reply source/manual.lisp]: Yes.

File: source/manual.lisp (hunk 3 above)
[the remark this answers is not in the document]
Comment (aartaka) [reply source/manual.lisp:16]: Ouch, sorry :)

File: source/tutorial.lisp (hunk 2 above)
Comment (aartaka) [reply source/tutorial.lisp]: Maybe double-colon?

But then, exporting those would be nice too...

Comment (aartaka) [reply source/tutorial.lisp]: Note that `url` is a generic function defined in `:nyxt` package, which might cause it too.

File: source/manual.lisp (hunk 3 above)
[the remark this answers is not in the document]
Comment (aadcg) [reply source/manual.lisp:16]: I had the same thought when I was looking a name!

File: source/tutorial.lisp (hunk 2 above)
Comment (aadcg) [reply source/tutorial.lisp]: Solved.

File: source/tutorial.lisp (hunk 1 above)
Comment (aadcg) [reply source/tutorial.lisp:145]: I will as soon as #2457 is closed, probably in another PR.

**Comment by aadcg** [comment]
Ok to merge @aartaka?

**Pull request merged**